### PR TITLE
fix go get subpackage bug

### LINKED ISSUE #2577: go-get meta tags are incorrect, cause infinite recursion
Here's a site running gitea: https://zxq.co/ripple/hanayo/routers/oauth?go-get=1. It says:

    <meta name="go-import" content="zxq.co/ripple/hanayo/routers/oauth git https://zxq.co/ripple/hanayo.git">

The general form of this tag is

    <meta name="go-import" content="IMPORT VCS REPO">

which says that the code at the root directory of VCS REPO is what to use for IMPORT. The tag above is incorrect because it says that https://zxq.co/ripple/hanayo.git is what to use for zxq.co/ripple/hanayo/routers/oauth, but in fact it's what to use for zxq.co/ripple/hanayo.

The correct tag is:

    <meta name="go-import" content="zxq.co/ripple/hanayo git https://zxq.co/ripple/hanayo.git">

That is, even though we asked about zxq.co/ripple/hanayo/routers/oauth, the meta tag should tell about the root of the git repo not the current subdirectory (the subdirectory will be deduced from the root).

This happens because [routers/repo/http.go](https://github.com/go-gitea/gitea/blob/master/routers/repo/http.go) includes the subdir in:

```
func composeGoGetImport(owner, repo, sub string) string {
	return path.Join(setting.Domain, setting.AppSubURL, owner, repo, sub)
}
```

I believe it would fix this problem to simply drop the ", sub" from the call.

### PR DESCRIPTION
**Pull request by lunny**
will fix #2577

### CODE REVIEW

**Comment by lafriks** [comment]
LGTM

File: routers/repo/http.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 )
 
 func composeGoGetImport(owner, repo, sub string) string {
```
Comment (daviian) [inline routers/repo/http.go]: Please remove unused parameter `sub`

**Review by daviian — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@daviian done.

**Comment by daviian** [comment]
LGTM

**Review by daviian — APPROVED** [review]

**Comment by sapk** [comment]
This is not working on try.gitea.io (bae9cbc) but at first glance don't know why since it should :disappointed: 
https://try.gitea.io/Howl/gorepo/test/test?go-get=1
```
<html>
	<head>
		<meta name="go-import" content="localhost/Howl/gorepo/test/test git https://try.gitea.io/Howl/gorepo.git">
		<meta name="go-source" content="localhost/Howl/gorepo/test/test _ https://try.gitea.io/Howl/gorepo/src/master{/dir} https://try.gitea.io/Howl/gorepo/src/master{/dir}/{file}#L{line}">
	</head>
	<body>
		go get localhost/Howl/gorepo/test/test
	</body>
</html>```

**Pull request merged**